src/audio/Kconfig

```diff
@@ -199,7 +199,7 @@ config COMP_IIR
 	bool "IIR component"
 	select COMP_BLOB
 	default y
-	depends on COMP_MODULE_ADAPTER
+	depends on COMP_MODULE_ADAPTER || DTS_CODEC
```

Comment:
I think this is the wrong way. I'd be happy with your original solution and just have this in the overlay as this is not a strict dependency at code level, but a requirement at the topologoy level, so I think the overlay is a better place.

Reply:
I also think this is not a good place, IIR can't depend on DTS, it is a standalone feature.

Reply:
@lgirdwood May I know your feedback?

Reply:
@joechengxperi I agree, I am not sure why the config is defined this way, you should `select` the IIR component from the DTS component.

Reply:
Thanks @cujomalainey. It looks better using select as you mentioned in this case. But after thinking about this a few days, I prefer to move it back to overlay rather than config. As @kv2019i said, it's not a strict dependency. Also, this dependency may probably change in the future, depends on customer requirement or platform. It seems using overlay and topology is more flexible.

Reply:
They should not depend on each other.
Besides I don't think this meets your intention "make sure IIR is enabled on different platforms when DTS_CODEC is enabled", which is "if DTS=y; set IIR=y". If so, you should state "config DTS select IIR"

Reply:
Quoting another thread:
> We use EQ IIR + DTS to support customer project.

Overlays is where you define that, not in Kconfig files (and overlays don't have to be in this git repo, they can come from anywhere).

The purpose of Kconfig is to make sure the code compiles and does not crash and nothing else (and managing these basic dependencies is difficult and error-prone enough already, random Kconfig problem of the day: #8425). Kconfig is only for "hard science" and universal dependencies that affect _everyone_ using the code base, never for "preferences" and even less for "personal" preferences. 

While Kconfig came from the Linux kernel, Zephyr has IMHO better Kconfig documentation:

https://docs.zephyrproject.org/latest/build/kconfig/index.html
https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html

They should be bug-for-bug compatible so both documentations should apply.

I see the current version does not seem to change Kconfig
https://github.com/thesofproject/sof/commits/1ef2f59db0a

Reply:
@marc-hb I removed that commit which added the wrong dependency and restored the original solution in overlay yesterday.